src/utils/rtl.tsx

```diff
@@ -180,3 +181,111 @@ function getTextFromTableCellNode(node: ChildNode) {
 export const withBeamRTL: Wrapper = {
   wrap: (c) => <BeamProvider>{c}</BeamProvider>,
 };
+
+/** Selects an option from the Beam SelectField, MultiSelectField, and TreeSelectField components
+ * For select fields that support multiple selections, subsequent calls to this function will toggle the selection state of an option.
+ * @param value The value or label of the option.
+ * */
+export function select(element: HTMLElement, value: string | string[]) {
+  const select = resolveIfNeeded(element);
+  if (isSelectElement(select)) {
+    throw new Error("Beam select helper does not support <select> elements.");
+  }
+
+  if (!isInputElement(select)) {
+    throw new Error(`Expected element to be INPUT, but got ${select.nodeName}. This field may be read-only.`);
+  }
+
+  function maybeOpenAndSelect(optionValue: string) {
+    const listboxId = select.getAttribute("aria-controls");
+    const body = select.closest("body") as HTMLElement;
+    const listbox = body.querySelector(`#${listboxId}`) as HTMLElement;
+    if (!listbox) {
+      _click(select);
+    }
+    const options: NodeListOf<HTMLElement> = body.querySelectorAll("[role=option]");
+    // Allow searching for options by their data-key (value) or textContent (label)
+    const optionToSelect = Array.from(options).find(
+      (o: HTMLElement) => o.dataset.key === optionValue || o.dataset.label === optionValue,
+    );
+    if (!optionToSelect) {
+      throw new Error(`Could not find option with value or text content of ${optionValue}`);
+    }
+    _click(optionToSelect);
+  }
+
+  const optionValues = Array.isArray(value) ? value : [value];
+  optionValues.forEach((optionValue) => {
+    maybeOpenAndSelect(optionValue);
+  });
+}
+
+export function selectAndWait(input: HTMLElement, value: string | string[]): Promise<void> {
+  return allowAndWaitForAsyncBehavior(() => select(input, value));
+}
+
+export function getSelected(element: HTMLElement): string[] | string | undefined {
+  const select = resolveIfNeeded(element);
+  if (isSelectElement(select)) {
+    throw new Error("Beam getSelected helper does not support <select> elements");
+  }
+
+  if (!isInputElement(select) && select.dataset.readonly === "true") {
+    // For read-only fields that render as a 'div'
+    return select.textContent ?? undefined;
+  }
+
+  const listboxId = select.getAttribute("aria-controls");
+  const body = select.closest("body") as HTMLElement;
+  const listbox = body.querySelector(`#${listboxId}`) as HTMLElement;
+  if (!listbox) {
+    _click(select);
+  }
+  const options: NodeListOf<HTMLElement> = body.querySelectorAll("[role=option]");
```

Comment:
Question: If we have two `SelectField`s on the page, is this going to bring back selections from both select fields?

Reply:
It should only return this selectfield's options, as the other select fields shouldn't be opened... but might be safer to query from the listbox rather than the body here. I'll update that.

Reply:
Oh I see, sure only one will be open at once, but yep thanks for updating!